File: python/maxProfit.py
class Solution:
    def maxProfit(prices: list[int]):
        
        to_buy = prices[0]
        best_profit = 0
        
        for i in prices[1:]:
            
            if i < to_buy:
                to_buy = i
            elif best_profit < i - to_buy:
                best_profit = i - to_buy
        
        print(best_profit)
            
                
            
                
                
        
        
        
        
        
      
        
solve = Solution.maxProfit([2,4,1])
            
        
       
                
        
        
        

File: C#/maxProfit.cs
class Solution {

    public int MaxProfit (int [] prices) {

        int toBuy = prices[0];

        int bestProfit = 0;

        for ( int i = 1; i < prices.Length; i++ ) {
            if ( prices[i] < toBuy) {
                toBuy = prices[i];
            } else if  (bestProfit < prices[i]- toBuy){ 
                bestProfit = prices[i]-toBuy;
            }
        }

        return bestProfit;
    }
}